services/terrain/src/terrain/clients/ezid.clj

```diff
@@ -0,0 +1,84 @@
+(ns ^{:author psarando} terrain.clients.ezid
+  (:use [ring.util.http-response :only [charset]]
+        [slingshot.slingshot :only [try+ throw+]])
+  (:require [cemerick.url :as curl]
+            [clj-http.client :as http]
+            [clojure.string :as string]
+            [terrain.util.config :as config]))
+
+(defn- auth-params
+  []
+  [(config/ezid-username) (config/ezid-password)])
+
+(defn- ezid-uri
+  [& components]
+  (str (apply curl/url (config/ezid-base-url) components)))
+
+(defn- anvl-escape
+  [value]
+  (-> value
+      (string/replace #"%"  "%25")
+      (string/replace #"\n" "%0A")
+      (string/replace #"\r" "%0D")
+      (string/replace #":"  "%3A")))
+
+(defn- anvl-unescape
+  [value]
+  (-> value
+      (string/replace #"%3A" ":")
+      (string/replace #"%0D" "\r")
+      (string/replace #"%0A" "\n")
+      (string/replace #"%25" "%")))
+
+(defn- anvl-decode
+  "Decodes an ANVL string into a map."
+  [anvl]
+  (let [anvl-lines (when (string? anvl) (->> anvl string/split-lines (remove string/blank?)))
+        keywordize-fn #(update-in % [0] keyword)]
+    (if (and anvl-lines (every? #(.contains % ":") anvl-lines))
+      (into {}
+        (map (fn [kv] (keywordize-fn (mapv anvl-unescape (string/split kv #": *" 2))))
+             anvl-lines))
```

Comment:
Er, hm. Looking again I realize your way is better, since it means you're not creating a map and then recreating the map with `keywordize-keys`, just creating it right the first time. It is a bit harder to read, but that's probably no big deal.